planning/behavior_path_planner/autoware_behavior_path_planner_common/src/utils/utils.cpp

```diff
@@ -648,6 +648,56 @@ double getDistanceToEndOfLane(const Pose & current_pose, const lanelet::ConstLan
   return lanelet_length - arc_coordinates.length;
 }
 
+double getDistanceToLastFitWidth(
+  const Pose & current_pose, const lanelet::ConstLanelets & lanelets, const double width_threshold)
+{
+  if (lanelets.empty()) return 0.0;
+  const double distance_to_end_of_lane = getDistanceToEndOfLane(current_pose, lanelets);
+  const double distance_from_last_fit_width_to_end =
+    getDistanceFromLastFitWidthToEnd(lanelets.back(), width_threshold);
+  return distance_to_end_of_lane - distance_from_last_fit_width_to_end;
+}
+
+double getDistanceFromLastFitWidthToEnd(
+  const lanelet::ConstLanelet & lane, const double width_threshold)
+{
+  const auto center_line = lane.centerline3d().basicLineString();
+  double distance = 0.0;
+
+  if (center_line.size() <= 1) return distance;
+
+  auto checkWidth = [&lane, &width_threshold](const Point & point) {
+    const double left_width = std::abs(getSignedDistanceFromLaneBoundary(lane, point, true));
+    const double right_width = std::abs(getSignedDistanceFromLaneBoundary(lane, point, false));
+    return (left_width > 0.5 * width_threshold && right_width > 0.5 * width_threshold);
+  };
+
+  auto it = center_line.rbegin() + 1;
+  Point point, next_point;
+  for (; it < center_line.rend(); ++it) {
+    point = lanelet::utils::conversion::toGeomMsgPt(*it);
+    next_point = lanelet::utils::conversion::toGeomMsgPt(*std::prev(it));
+    distance += autoware::universe_utils::calcDistance2d(point, next_point);
+    if (checkWidth(point)) break;
+  }
+
+  constexpr double step_length{1.0};
+  const double seg_length = autoware::universe_utils::calcDistance2d(point, next_point);
+  if (seg_length <= step_length) return distance;
+
+  double yaw = atan2(next_point.y - point.y, next_point.x - point.x);
+  double length = step_length;
+  for (; length < seg_length; length += step_length) {
+    auto mid_point = point;
+    mid_point.x += length * cos(yaw);
+    mid_point.y += length * sin(yaw);
```

Comment:
I don't think you need to calculate the yaw here. You can just interpolate between the two points using `universe_utils::calcInterpolatedPoint` (https://github.com/autowarefoundation/autoware.universe/blob/main/common/autoware_universe_utils/include/autoware/universe_utils/geometry/geometry.hpp#L472).

Alternatively, how about resampling the centerline with a fixed interval before looping over its points ?

Reply:
I don't think its necessary to resample the entire centerline, its more efficient to interpolate the segment where condition applies